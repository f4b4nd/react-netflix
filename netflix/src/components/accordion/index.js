import { Container, Frame, Title, Item, Inner, Header, Body } from './styles/accordion' 
import { useState, useContext, createContext } from 'react'

export default function Accordion ({children, ...restProps}) {
    return (
        <Container {...restProps} > 
            <Inner> {children} </Inner>
        </Container>
    )
}

/**GENERIC*/
Accordion.Inner = function AccordionInner({children, ...restProps}) {
    return <Inner {...restProps} > {children} </Inner>
}

Accordion.Title = function AccordionTitle({children, ...restProps}) {
    return <Title {...restProps} > {children} </Title>
}

Accordion.Frame = function AccordionFrame({children, ...restProps}) {
    return <Frame {...restProps} > {children} </Frame>
}

/**MAP*/
const ToggleContext = createContext()

Accordion.Item = function AccordionItem({children, ...restProps}) {
    const [toggleShow, setToggleShow] = useState(false)
    return (
        <ToggleContext.Provider value={{toggleShow, setToggleShow}}> 
            <Item {...restProps} > {children} </Item>
        </ToggleContext.Provider>
    )
}

Accordion.Header = function AccordionHeader({children, ...restProps}) {
    const {toggleShow, setToggleShow} = useContext(ToggleContext)
    return (        
        <Header 
            {...restProps} 
            onClick={() => setToggleShow(toggleShow => !toggleShow)}
        > 
            {children} 
        </Header>
    )
}

Accordion.Body = function AccordionFrame({children, ...restProps}) {
    const {toggleShow, setToggleShow} = useContext(ToggleContext)
    if (toggleShow) {    
        return <Body {...restProps} > {children} </Body>
    } else {
        return null
    }
}