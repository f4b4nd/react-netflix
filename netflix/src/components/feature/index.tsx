import { Container, Title, SubTitle } from './style'

export default function Feature({ children, ...restProps }: IChildren) {
    return <Container {...restProps}>{children}</Container>
}

Feature.Title = function FeatureTitle({ children, ...restProps }: IChildren) {
    return <Title {...restProps}>{children}</Title>;
}

Feature.SubTitle = function FeatureSubTitle({ children, ...restProps }: IChildren) {
    return <SubTitle {...restProps}>{children}</SubTitle>
}