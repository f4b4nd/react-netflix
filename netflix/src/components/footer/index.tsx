import { Container, Row, Column, Link, Title, Text, Break } from "./style"

export default function Footer ({children, ...restProps}: IChildren)  {
    return (
        <Container {...restProps}> {children} </Container>
    )
}
  
Footer.Row = function FooterRow({ children, ...restProps }: IChildren) {
    return <Row {...restProps}> {children} </Row>
}

Footer.Column = function FooterColumn({ children, ...restProps }: IChildren) {
    return <Column {...restProps}> {children} </Column>
}

Footer.Link = function FooterLink({ children, ...restProps }: IFooterLink) {
    return <Link {...restProps}> {children} </Link>
}

Footer.Title = function FooterTitle({ children, ...restProps }: IChildren) {
    return <Title {...restProps}> {children} </Title>
}

Footer.Text = function FooterText({ children, ...restProps }: IChildren) {
    return <Text {...restProps}> {children}</Text>
}

Footer.Break = function FooterBreak({ ...restProps }: IChildren) {
    return <Break {...restProps} />
}