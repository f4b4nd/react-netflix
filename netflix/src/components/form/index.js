import { Container, Title, Email, Password, Button } from "./styles/form"

export default function Form ({children, ...restProps}) {


    return (
        <Container> 
            {children} 
        </Container>
    )
}

Form.Title = function FormTitle({children, ...restProps}) {
    return <Title {...restProps}> {children} </Title>
}

Form.Email = function FormEmail({children, ...restProps}) {
    

    return <Email {...restProps} />
}

Form.Password = function FormPassword({children, ...restProps}) {

    return <Password {...restProps} />
    
}

Form.Button = function FormButton({children, ...restProps}) {

    return (
        <Button {...restProps}> 
            {children} 
        </Button>
    )
    
}

