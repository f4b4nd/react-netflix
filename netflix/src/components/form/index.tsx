
import { Container, Error, Base, Title, Text, TextSmall, Link, Input, Submit } from './style'

export default function Form({ children, ...restProps }: IChildren) {
    return <Container {...restProps}>{children}</Container>
}

Form.Error = function FormError({ children, ...restProps }: IChildren) {
    return <Error {...restProps}>{children}</Error>
}

Form.Base = function FormBase({ children, ...restProps }: IFormBase) {
    return <Base {...restProps}>{children}</Base>
}

Form.Title = function FormTitle({ children, ...restProps }: IChildren) {
    return <Title {...restProps}>{children}</Title>
}

Form.Text = function FormText({ children, ...restProps }: IChildren) {
    return <Text {...restProps}>{children}</Text>
}

Form.TextSmall = function FormTextSmall({ children, ...restProps }: IChildren) {
    return <TextSmall {...restProps}>{children}</TextSmall>
}

Form.Link = function FormLink({ children, ...restProps }: IFormLink) {
    return <Link {...restProps}>{children}</Link>
}

Form.Input = function FormInput({ children, ...restProps }: IFormInput) {
    return <Input {...restProps}>{children}</Input>
}

Form.Submit = function FormSubmit({ children, ...restProps }: IFormSubmit) {
    return <Submit {...restProps}>{children}</Submit>
}