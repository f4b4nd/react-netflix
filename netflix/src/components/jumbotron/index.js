import { Container, Inner } from "./styles/jumbotron"

export default function Jumbotron ({children, direction = 'column', ...restProps}) {
    return (
        <Inner direction={direction} >
            {children}
        </Inner>
    )
}

Jumbotron.Container = function JumbotronContainer({children, ...restProps}) {
    return (
        <Container {...restProps}> {children} </Container>
    )
}
