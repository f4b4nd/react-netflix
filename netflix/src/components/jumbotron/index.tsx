import { Container, Item, Inner, Pane, Title, SubTitle, Image  } from "./style"

export default function Jumbotron ({children, flexDirection = 'column', ...restProps}: IJumbotron) {
    return (
        <Item {...restProps} >
            <Inner flexDirection={flexDirection}> 
                {children}
            </Inner>
        </Item>
    )
}

Jumbotron.Container = function JumbotronContainer({children, ...restProps}: IChildren) {
    return (
        <Container {...restProps}> {children} </Container>
    )
}

Jumbotron.Pane = function JumbotronPane({children, ...restProps}: IChildren) {
    return (
        <Pane {...restProps}> {children} </Pane>
    )
}

Jumbotron.Title = function JumbotronTitle({children, ...restProps}: IChildren) {
    return (
        <Title {...restProps}> {children} </Title>
    )
}

Jumbotron.SubTitle = function JumbotronSubTitle({children, ...restProps}: IChildren) {
    return (
        <SubTitle {...restProps}> {children} </SubTitle>
    )
}

Jumbotron.Image = function JumbotronImage({...restProps}) {
    return (
        <Image {...restProps} />
    )
}