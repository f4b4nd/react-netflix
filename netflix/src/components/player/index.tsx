import { useState, useContext,  } from 'react'

import ReactDOM from 'react-dom'

import { Container, Button, Overlay, Inner, Close } from './style'

import { PlayerContext } from '../../context/player'

export default function Player({ children, ...restProps }: IChildren) {

    const [showPlayer, setShowPlayer] = useState<boolean>(false)

    return (
        <PlayerContext.Provider value={{ showPlayer, setShowPlayer }}>
            <Container {...restProps}>
                {children}
            </Container>
        </PlayerContext.Provider>
    )

}

Player.Video = function PlayerVideo({ src }: IPlayerVideo) {

    const { showPlayer, setShowPlayer } = useContext(PlayerContext)

    return (
        showPlayer ? (
            ReactDOM.createPortal(
                <Overlay onClick={() => setShowPlayer(false)} data-testid="player">
                    <Inner>
                        <video id="player" controls>
                            <source src={src} type="video/mp4" />
                        </video>
                        <Close />
                    </Inner>bodybody
                </Overlay>,
                document.body
            )
        ) : null
    )
}

Player.Button = function PlayerButton({ ...restProps }) {

    const { showPlayer, setShowPlayer } = useContext(PlayerContext)

    return (
        <Button 
            onClick={() => setShowPlayer(!showPlayer)} 
            {...restProps}
        >
            Play
        </Button>
    )
}