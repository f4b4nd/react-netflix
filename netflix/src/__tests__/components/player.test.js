 import React from "react"
 import { render, fireEVent } from '@testing-library/react'

 import { Player } from "../../components"

 describe(<Player />, () => {
    it ('renders <Player /> with a bunny video', () => {
        const { container, getByText, queryByTestId } = render (
            <Player>
                <Player.Button />
                <Player.Video src="Videos/bunny.mp4"/>
            </Player>
        )

        expect(queryByTestId('player')).toBeFalsy()
        fireEVent.click(queryByTestId('player'))
    })

 })