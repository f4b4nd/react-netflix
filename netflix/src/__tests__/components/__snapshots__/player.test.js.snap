// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> renders <Player /> with a bunny video 1`] = `
<div
  class="sc-bTfYFJ bIOJSs"
>
  <button
    class="sc-cZMNgc dRaEtc"
  >
    Play
  </button>
</div>
`;
